File: admin_portal/backend/schemas/provider.py
"""Provider schemas."""
from typing import List, Optional
from pydantic import BaseModel


class ProviderCreate(BaseModel):
    name: str
    aws_region: str
    auth_type: str  # "ak_sk" or "bearer_token"
    access_key_id: Optional[str] = None
    secret_access_key: Optional[str] = None
    bearer_token: Optional[str] = None
    endpoint_url: Optional[str] = None


class ProviderUpdate(BaseModel):
    name: Optional[str] = None
    aws_region: Optional[str] = None
    auth_type: Optional[str] = None
    access_key_id: Optional[str] = None
    secret_access_key: Optional[str] = None
    bearer_token: Optional[str] = None
    endpoint_url: Optional[str] = None
    is_active: Optional[bool] = None


class ProviderResponse(BaseModel):
    provider_id: str
    name: str
    aws_region: str
    auth_type: str
    has_access_key: bool = False
    has_secret_access_key: bool = False
    has_bearer_token: bool = False
    endpoint_url: Optional[str] = None
    is_active: bool = True
    created_at: int = 0
    updated_at: int = 0


class ProviderListResponse(BaseModel):
    items: List[ProviderResponse]
    count: int

File: admin_portal/frontend/src/types/provider.ts
export interface Provider {
  provider_id: string;
  name: string;
  aws_region: string;
  auth_type: 'ak_sk' | 'bearer_token';
  has_access_key: boolean;
  has_secret_access_key: boolean;
  has_bearer_token: boolean;
  endpoint_url?: string;
  is_active: boolean;
  created_at: number;
  updated_at: number;
}

export interface ProviderCreate {
  name: string;
  aws_region: string;
  auth_type: 'ak_sk' | 'bearer_token';
  access_key_id?: string;
  secret_access_key?: string;
  bearer_token?: string;
  endpoint_url?: string;
}

export interface ProviderUpdate {
  name?: string;
  aws_region?: string;
  auth_type?: 'ak_sk' | 'bearer_token';
  access_key_id?: string;
  secret_access_key?: string;
  bearer_token?: string;
  endpoint_url?: string;
  is_active?: boolean;
}

export interface ProviderListResponse {
  items: Provider[];
  count: number;
}
